### PR TITLE
feat: increase reveal price

### PR DESCRIPTION
**Pull request by metacertain**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
To make sure reveal transactions are processed in time with higher reliability, increase suggested gasprice by 50%
Gas limits are also lowered for commit and reveal to help the transactions being processed more easily

### Screenshots (if appropriate):

- - -
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3452)

### CODE REVIEW

**Comment by ldeffenb** [comment]
Ready and anxious for this to get merged!   I've had so many stakes slashed because of gas prices increasing causing my reveal to be delayed!   You might also want to do it for the claim, which granted only causes lost reward, but I've still had those delayed as well.   Possibly make the percentage boosts configurable in the long term?   I was actually thinking 15-30% boost on the reveal and 10-20% boost on the claim.   I've been monitoring the gas price change percentages via my up-and-coming monSI (https://github.com/ldeffenb/monSI).

File: pkg/storageincentives/redistribution/redistribution.go [hunk 1]
```diff
@@ -149,8 +149,8 @@ func (c *contract) Reveal(ctx context.Context, storageDepth uint8, reserveCommit
 	request := &transaction.TxRequest{
 		To:          &c.incentivesContractAddress,
 		Data:        callData,
-		GasPrice:    sctx.GetGasPrice(ctx),
-		GasLimit:    sctx.GetGasLimitWithDefault(ctx, 3_000_000),
+		GasPrice:    new(big.Int).Div(new(big.Int).Mul(sctx.GetGasPrice(ctx), big.NewInt(15)), big.NewInt(10)),
```
Comment (aloknerurkar) [inline pkg/storageincentives/redistribution/redistribution.go:152]: So what is this calculation doing? Can we add some comments?

Comment (ldeffenb) [reply pkg/storageincentives/redistribution/redistribution.go:152]: Actually, what happens if sctx.GetGasPrice return nil?  If the context doesn't have a gas-price key, that's what will be coming back.   And I failed to track down where ctx actually comes from, but I'm thinking this multiplicative factor is actually in the wrong place, but needs to be gasPrice has been resolved at https://github.com/ethersphere/bee/blob/00aada8a51a06e2997ba7a4f1df6b717c853d99e/pkg/transaction/transaction.go#L276

**Review by notanatol — APPROVED** [review]

File: pkg/transaction/transaction.go [hunk 2]
```diff
@@ -293,6 +355,47 @@ func (t *transactionService) prepareTransaction(ctx context.Context, request *Tx
...
+		gasLimit += gasLimit / 5 // add 20% on top
+
+	} else {
+		gasLimit = request.GasLimit
+	}
+
+	gasPrice := request.GasPrice
+	if gasPrice == nil {
+		gasPriceSuggested, err := t.backend.SuggestGasPrice(ctx)
+		gasPrice = new(big.Int).Div(new(big.Int).Mul(big.NewInt(int64(boostPercent)+100), gasPriceSuggested), big.NewInt(100))
```
Comment (ldeffenb) [inline pkg/transaction/transaction.go:380]: This is exactly where I was envisioning the boost needed to be!

**Review by notanatol — APPROVED** [review]

File: pkg/transaction/transaction.go [hunk 3]
```diff
@@ -199,6 +201,66 @@ func (t *transactionService) Send(ctx context.Context, request *TxRequest) (txHa
 	return signedTx.Hash(), nil
 }
 
+// Send creates and signs a transaction based on the request and sends it.
+func (t *transactionService) SendWithBoost(ctx context.Context, request *TxRequest, boostPercent uint64) (txHash common.Hash, err error) {
```
Comment (istae) [inline pkg/transaction/transaction.go:205]: it'd be great if you can write a simple test that checks these "boosted" prices

**Review by istae — CHANGES_REQUESTED** [review]

**Comment by agazso** [comment]
At this point it would make a lot of sense to investigate using [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559) which changes how gas prices are estimated .

**Comment by ldeffenb** [comment]
> At this point it would make a lot of sense to investigate using [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559) which changes how gas prices are estimated .

But please allow this update to merge to at least provide some mitigation on the testnet.   I'm cleaning up a node right now that has 6 backlogged transactions due to the daily upward spiral of goerli gas prices.

**Comment by istae** [comment]
> At this point it would make a lot of sense to investigate using [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559) which changes how gas prices are estimated .

@agazso we've had this PR hanging around for a while https://github.com/ethersphere/bee/pull/3021

**Review by aloknerurkar — APPROVED** [review]

**Review by istae — APPROVED** [review]

**Pull request merged**